modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcTransactionTest.java

```diff
@@ -0,0 +1,307 @@
+     * <ul>
+     * <li>Auto-commit mode is enabled by default.
+     * <li>If {@code setAutoCommit} is called and the auto-commit mode is not changed, the call is a no-op.
+     * <li>If {@code setAutoCommit} is called during transaction and auto-commit mode changed then active transaction is committed.
+     * <li>Call to {@code setAutoCommit} or {@code getAutoCommit} on a closed connection throws exception.
+     * </ul>
+     *
+     * @throws SQLException If failed.
+     */
+    @Test
+    public void testAutoCommitModeChange() throws SQLException {
+        try (Connection conn = DriverManager.getConnection(URL)) {
+            assertTrue(conn.getAutoCommit());
+            conn.setAutoCommit(true);
+
+            conn.setAutoCommit(false);
+            conn.setAutoCommit(false);
+
+            int expRowsCount = 3;
+            String sqlUpdate = "insert into TEST (ID) values (1), (2), (3)";
+
+            try (Statement stmt = conn.createStatement()) {
+                conn.setAutoCommit(false);
+
+                assertEquals(expRowsCount, stmt.executeUpdate(sqlUpdate));
+                assertEquals(expRowsCount, rowsCount(conn));
+                assertEquals(0, rowsCount(AbstractJdbcSelfTest.conn));
+
+                // Verify that switching to auto-commit mode again commits started transaction.
+                conn.setAutoCommit(true);
+                assertEquals(expRowsCount, rowsCount(AbstractJdbcSelfTest.conn));
+            }
+
+            conn.close();
+
+            // Exception when called on closed connection
+            checkConnectionClosed(conn::getAutoCommit);
+            checkConnectionClosed(() -> conn.setAutoCommit(true));
+            checkConnectionClosed(() -> conn.setAutoCommit(false));
+        }
+    }
+
+    /**
+     * Tests {@link Connection#commit()} method behaviour.
+     *
+     * <p>Calling {@code commit} is expected to throw an exception if called in auto-commit mode or on a closed connection,
+     *
+     * @throws SQLException If failed.
+     */
+    @Test
+    public void testCommit() throws Exception {
+        try (Connection conn = DriverManager.getConnection(URL)) {
+            SQLException ex = assertThrows(SQLException.class, conn::commit);
+            MatcherAssert.assertThat(ex.getMessage(), containsString("Transaction cannot be committed explicitly in auto-commit mode."));
+
+            conn.setAutoCommit(false);
+            conn.commit();
+
+            conn.close();
+
+            // Exception when called on closed connection
+            checkConnectionClosed(conn::commit);
+        }
+    }
+
+    /**
+     * Tests {@link Connection#rollback()} method behaviour.
+     *
+     * <p>Calling {@code rollback} is expected to throw an exception if called in auto-commit mode or on a closed connection,
+     *
+     * @throws SQLException If failed.
+     */
+    @Test
+    public void testRollback() throws Exception {
```

Comment:
this one should be unit as well

Reply:
Test has been moved to `JdbcQueryEventHandlerImplTest`